src/plugin.py

```diff
@@ -31,7 +31,7 @@ def before_promote(self, conn, config):
         """
         pass
 
-    def after_promote(self, conn, config):
+    def after_promote(self, conn, wals_count_to_upload: int):
```

Comment:
And that is wrong: generic PostgresPlugin is not limited to deal with `wals_count_to_upload` only. It can perform some other actions, and entire config is more useful for it here.